gobblin-data-management/src/main/java/gobblin/data/management/retention/policy/NoOpRetentionPolicy.java

```diff
@@ -0,0 +1,41 @@
+/*
+* Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+* this file except in compliance with the License. You may obtain a copy of the
+* License at  http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software distributed
+* under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+* CONDITIONS OF ANY KIND, either express or implied.
+*/
+
+package gobblin.data.management.retention.policy;
+
+import java.util.Collection;
+import java.util.List;
+
+import azkaban.utils.Props;
+
+import gobblin.data.management.retention.version.DatasetVersion;
+import gobblin.data.management.retention.version.StringDatasetVersion;
+
+/**
+ * Implementation of {@link RetentionPolicy} that uses a {@link StringDatasetVersion} and marks all
+ * {@link StringDatasetVersion}s as deletable.
+ */
+public class NoOpRetentionPolicy implements RetentionPolicy<StringDatasetVersion> {
```

Comment:
I would not call this NoOp. To me NoOp means nothing will be deleted. Maybe call it DeleteAllRetentionPolicy instead?